src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -50,18 +50,24 @@ public void RequestStart(HttpRequestMessage request)
                 request.VersionPolicy);
         }
 
-        [Event(2, Level = EventLevel.Informational)]
-        public void RequestStop()
+        [NonEvent]
+        public void RequestStop(HttpResponseMessage? response)
+        {
+            RequestStop(response is null ? -1 : (int)response.StatusCode);
```

Comment:
Does a null `response` mean that the request didn't even get sent, or that a response was never received?

Reply:
It means that the response object didn't make it up to the caller (HttpClient), or in other words because the call to `SendAsync` threw an exception.
And that can happen at any stage of a request, e.g. we timed out before even getting a connection, something blew up after we sent the request but before we got a response, we failed to parse the response, user code in a custom handler somewhere threw an exception ...